src/validate/mod.rs

```diff
@@ -16,6 +16,9 @@ lazy_static! {
   static ref SENDER_FORMAT: Regex = Regex::new(
     "^[A-Za-z0-9-]+(?: [A-Za-z0-9-]+)* <[a-z0-9-]+@[a-z0-9-]+(?:\\.[a-z0-9-]+)+>$"
   ).unwrap();
+  static ref BASE_URI_FORMAT: Regex = Regex::new(
+    "^https?://[A-Za-z0-9-]+(?:\\.[A-Za-z0-9-]+)*(?::[0-9]+)?/(?:[A-Za-z0-9-]+/)*$"
```

Comment:
This regex is overly-restrictive as a general base URI format, but it's correct for our usage.